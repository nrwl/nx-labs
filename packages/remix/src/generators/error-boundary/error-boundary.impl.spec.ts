import { addProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import errorBoundaryGenerator from './error-boundary.impl';

describe('ErrorBoundary', () => {
  describe('--apiVersion=1', () => {
    it('should correctly add the ErrorBoundary to the route file', async () => {
      // ARRANGE
      const tree = createTreeWithEmptyWorkspace();
      addProjectConfiguration(tree, 'demo', {
        name: 'demo',
        root: '.',
        sourceRoot: '.',
        projectType: 'application',
      });
      const routeFilePath = `app/routes/test.tsx`;
      tree.write(routeFilePath, ``);
      tree.write('remix.config.js', `module.exports = {}`);

      // ACT
      await errorBoundaryGenerator(tree, {
        project: 'demo',
        path: routeFilePath,
        apiVersion: 1,
      });

      // ASSERT
      expect(tree.read(routeFilePath, 'utf-8')).toMatchSnapshot();
    });

    it('should error when the route file cannot be found', async () => {
      // ARRANGE
      const tree = createTreeWithEmptyWorkspace();
      addProjectConfiguration(tree, 'demo', {
        name: 'demo',
        root: '.',
        sourceRoot: '.',
        projectType: 'application',
      });
      const routeFilePath = `app/routes/test.tsx`;
      tree.write(routeFilePath, ``);
      tree.write('remix.config.js', `module.exports = {}`);

      // ACT & ASSERT
      await expect(
        errorBoundaryGenerator(tree, {
          project: 'demo',
          path: `my-route.tsx`,
          apiVersion: 1,
        })
      ).rejects.toThrow();
    });
  });

  describe('--apiVersion=2', () => {
    it('should correctly add the ErrorBoundary to the route file', async () => {
      // ARRANGE
      const tree = createTreeWithEmptyWorkspace();
      addProjectConfiguration(tree, 'demo', {
        name: 'demo',
        root: '.',
        sourceRoot: '.',
        projectType: 'application',
      });
      const routeFilePath = `app/routes/test.tsx`;
      tree.write(routeFilePath, ``);
      tree.write('remix.config.js', `module.exports = {}`);

      // ACT
      await errorBoundaryGenerator(tree, {
        project: 'demo',
        path: routeFilePath,
        apiVersion: 2,
      });

      // ASSERT
      expect(tree.read(routeFilePath, 'utf-8')).toMatchSnapshot();
    });

    it('should error when the route file cannot be found', async () => {
      // ARRANGE
      const tree = createTreeWithEmptyWorkspace();
      addProjectConfiguration(tree, 'demo', {
        name: 'demo',
        root: '.',
        sourceRoot: '.',
        projectType: 'application',
      });
      const routeFilePath = `app/routes/test.tsx`;
      tree.write(routeFilePath, ``);
      tree.write('remix.config.js', `module.exports = {}`);

      // ACT & ASSERT
      await expect(
        errorBoundaryGenerator(tree, {
          project: 'demo',
          path: `my-route.tsx`,
          apiVersion: 2,
        })
      ).rejects.toThrow();
    });
  });
});
