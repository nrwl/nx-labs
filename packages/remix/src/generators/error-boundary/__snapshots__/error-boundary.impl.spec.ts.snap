// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary --apiVersion=1 should correctly add the ErrorBoundary to the route file 1`] = `
"export function ErrorBoundary({ error }) {
  console.error(error);
  return (
    <div>
      <h1>Uh oh ...</h1>
      <p>Something went wrong</p>
      <pre>{error.message || 'Unknown error'}</pre>
    </div>
  );
}
"
`;

exports[`ErrorBoundary --apiVersion=2 should correctly add the ErrorBoundary to the route file 1`] = `
"import { isRouteErrorResponse, useRouteError } from '@remix-run/react';
export function ErrorBoundary() {
  const error = useRouteError();

  // when true, this is what used to go to 'CatchBoundary'
  if (isRouteErrorResponse(error)) {
    return (
      <div>
        <h1>Oops</h1>
        <p>Status: {error.status}</p>
        <p>{error.data.message}</p>
      </div>
    );
  }

  // Don't forget to typecheck with your own logic.
  // Any value can be thrown, not just errors!
  let errorMessage = 'Unknown error';
  // if (isDefinitelyAnError(error)) {
  //    errorMessage = error.message;
  // }

  return (
    <div>
      <h1>Uh oh ...</h1>
      <p>Something went wrong.</p>
      <pre>{errorMessage}</pre>
    </div>
  );
}
"
`;
